### ISSUE TITLE
Petitioner: Modify E-filing Flow to Separate Attachments to Petition

### ISSUE THREAD

**Issue by cholly75** [ustaxcourt/ef-cms#8587]
As a petitioner, so that my case clearly reflects the petition and what I've attached to the petition, I need the Create a Case workflow to provide a specific place to upload attachments to my petition.

As the court moves towards greater transparency on its docket record, one aspect we want to address is that e-filed petitions often (at our direction) contain not only the petition but an unknown quantity of attachments to the petition, all in one PDF file.

In addition to lumping relevant and irrelevant documents (such as the IRS notice) into the same docket entry with the petition, this also creates questions from petitioners about how to merge PDF files, which documents should be combined, and other technical/process questions. These documents are also often the ones least redacted by petitioners, which will cause difficulty in moving towards greater transparency on the docket for petitions. 

## Pre-Conditions
ustaxcourt/ef-cms#8578 ustaxcourt/ef-cms#8587

## Acceptance Criteria
* Create a Case e-filing flow is updated to remove mentions of merging IRS notices with Petition PDF
* Create a Case e-filing flow provides a step to upload IRS notices separately from petition
  - It is not required to upload any notice files
  - Multiple notices can be uploaded separately in that step of the workflow
  - Notices abide by usual e-filing requirements (PDF, 250 MB max, etc)
* Uploaded notice files are displayed on final step of Create a Case workflow
* Notice files are submitted along with petition/STIN/CDS files in final step of the workflow
* Each notice file uploaded is entered as an individual ATP event into DAWSON upon submission
* ATPs are visible in petitions review by the clerk
  - If multiple ATP uploads were submitted, each upload is displayed to the clerk during QC
* ATPs submitted with petitions are served concurrently with the petition when reviewed

## Notes
* Do we need a maximum allowable number of Notice uploads to prevent abuse? - yes, 5 initially
* 

## Tasks
 - [x] Dashboard - Update "Other Ways to File" to Accordion 
 - [x] Dashboard - Update text to future planned petitions text
 - [x] Page 1 - Update 'Prepare the petition' accordion with new text
 - [x] Page 2 - Use USWDS to capture IRS Notices (limit to 5 files pdfs)
 - [x] Final Review - Move service email into the petitioner contact info section 
 - [x] Adding an IRS notice label with the IRS notice pdf linked (if they added one. If they didn't, then don't display the label either)
 - [x] Add ATP docs to Attachments Card
 - [x] Serve ATPs docs as individual docket entry
 - [x] Use document Id for document selections

## Story Definition of Ready (updated on 12/23/22)
The following criteria must be met in order for the user story to be picked up by the Flexion development team.
The user story must: 
- [ ] Is framed in business/user need, the value has been addressed.
- [ ] Includes acceptance criteria
- [ ] Has been refined
- [ ] Pre conditions have been satisfied. 

**Process:** 
Flexion developers and designers will test if the story meets acceptance criteria and test cases in Flexion dev and staging environments (“standard testing”). If additional acceptance criteria or testing scenarios are discovered while the story is in progress, a new story should be created, added to the backlog and prioritized by the product owner. 

## Definition of Done (Updated 5-19-22)
**Product Owner**
 - [ ]  Acceptance criteria have been met and validated on the Court's migration environment
 - [ ] Add scenario to testing document, if applicable (https://docs.google.com/spreadsheets/d/1FUHKC_YrT-PosaWD5gRVmsDzI1HS_U-8CyMIb-qX9EA/edit?usp=sharing)

**UX**
 - [ ] Business test scenarios have been refined to meet all acceptance criteria 
 - [ ] Usability has been validated
 - [ ] Wiki has been updated (if applicable) 
 - [ ] Story has been tested on a mobile device (for external users only)

**Engineering**
 - [x] Automated test scripts have been written, including visual tests for newly added PDFs.
 - [x] Field level and page level validation errors (front-end and server-side) integrated and functioning.
 - [x] Verify that language for docket record for internal users and external users is identical.
 - [x] New screens have been added to pa11y scripts.
 - [ ] All new functionality verified to work with keyboard and macOS voiceover https://www.apple.com/voiceover/info/guide/_1124.html.
 - [x] READMEs, other appropriate docs, and swagger/APIs fully updated.
 - [x] UI should be touch optimized and responsive for external only (functions on supported mobile devices and optimized for screen sizes as required).
 - [x] Interactors should validate entities before calling persistence methods.
 - [x] Code refactored for clarity and to remove any known technical debt.
 - [x] If new docket entries have been added as seed data to `efcms-local.json`, 3 local s3 files corresponding to that docketEntryId have been added to `web-api/storage/fixtures/s3/noop-documents-local-us-east-1`
 - [x] Acceptance criteria for the story has been met.
 - [x] If there are special instructions in order to deploy into the next environment, add them as a comment in the story.
 - [x] If the work completed for the story requires a reindex without a migration, or any other special deploy steps, apply these changes to the following flexion branches:
   - [ ] experimental1
   - [ ] experimental2
   - [ ] experimental3
   - [ ] experimental4
   - [ ] experimental5
   - [ ] experimental6
   - [ ] develop
 - [ ] Reviewed by UX on a deployed environment.
 - [ ] Reviewed by PO on a deployed environment. Can be deployed to the Court's `test` environment if prod-like data is required. Otherwise deployed to any `experimental` environment.
 - [ ] Deployed to the Court's `staging` environment.

*

**Comment by katiecissell** [comment]
UX Notes:

There are five pages that will need updating.

[Figma file](https://www.figma.com/file/6uihfDrN6nGBPonTdIZ8lq/Generate-Petition?type=design&node-id=717%3A4420&mode=design&t=Cu5EXwxId3uAmddv-1)

First time filer page will have its text overhauled to use the future planned petitions text. Also note the bottom right card has accordions added to it now. 
<img width="663" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBNndYQXc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--62fcbbd4d420b0840142b8be3ce915acf521f47d/Start%20page%20-%20first%20time%20filer.jpg" alt="Start page - first time filer.jpg" />

After the user clicks create case or start case, we will be updating the third accordion's text:
<img width="1131" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBM2VUQWc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--8c7d15443ca9f690cd7c4a5a5d8de07047a86b9a/accordion%20text.png" alt="accordion text.png" />

During the petition flow we will be updating step 2 so that if the user clicks yes there was an IRS notice, we will be adding the new USWDS upload component that allows for multiple uploads. We should limit it to 5.
<img width="802" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBL2tDQXc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--f184a732b902a0e0292012cb79f135811a3c2d7e/image.png" alt="image.png" />

And finally on the review step, we are moving the service email into the petitioner contact info section and adding an IRS notice label with the IRS notice pdf linked (if they added one. If they didn't, <s>then don't display the label either</s> display "N/A")
<img width="834" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBM2lUQWc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--195b62a2f0060281c5a4caf02cb7d360e26b0a7b/Review%20case%20upload.jpg" alt="Review case upload.jpg" />

During the QC process, each uploaded IRS notice should appear as a separate tab on the right side. 
<img width="1192" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBekdUQWc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--19a70f2fae3cdd69f5aec4b4b2a7d6f9ec196bd2/image.png" alt="image.png" />

**Comment by katiecissell** [comment]
Pre-refinement questions:
- Uploading multiple files would require the new upload component, are we switching to the new one here or keeping the old one which only allows for one file?
- if they upload multiple separate IRS notices, should the ATP tab in the QC process combine all of them into one file or keep them separate?

**Comment by ttlenard** [comment]
The mockup above shows two different upload components. Let's make sure that whichever one is selected is consistent throughout the Petition workflow.  We shouldn't have two different components displayed for the user.

**Comment by katiecissell** [comment]
@cholly75 referring to Tenille's comment above^, do we want to include the work to swap out the upload components for the STIN, Petition and CDS in this ticket or a different one?

**Comment by ttlenard** [comment]
## Test Cases

#### 1) First time e-Filer logs into DAWSON; Welcome page includes new text, including a section about including a PDF of the IRS Notice and accordion menus.

- Log in to DAWSON as a Petitioner that does not yet have any cases
- Review the Welcome screen.

Expected Results:
- [x] **NEW FUNCTIONALITY** - Text updates on this page match the mockup.
- [x] **NEW FUNCTIONALITY** - Information in the Other Ways to File card on the right side of the screen are now organized into accordion menus.



#### 2) Petitioner (first time filer AND any other Petitioner) clicks the Start a Case button, new instructions are on the Instructions For Creating a Case page.

- Continuing from the previous test
- Click on the Start a case button
- Expand the accordion menus

Expected Results:
- [x] **NEW FUNCTIONALITY** - The third accordion menu is now titled "Prepare the Petition and any IRS Notices"
- [x] **NEW FUNCTIONALITY** - The text that is in this expanded menu matches the text in the mockup.



#### 3) Petitioner (first time filer AND any other Petitioner) starts case, uploads STIN; New Section uploading IRS Notice is on Step 2 if user selects that they did receive an IRS Notice.

- Continuing from the previous test
- Click Got it, Let's Start a Case button
- Upload a STIN and click Continue
- Upload a Petition
- Check the Yes radio button under the Why are you filing this Petition? section

Expected Results:
- [x] **NEW FUNCTIONALITY** - If the Yes button is selected, a Upload PDF of IRS Notice(s) section displays.



#### 4) Petitioner can ~~drag a file or upload 1-5 PDF's into the Notice section~~  upload a singular IRS Notice

- Continuing from the previous test
- ~~Drag~~ Upload a single file from your computer into the IRS notice section

Expected Results:
- [x] **NEW FUNCTIONALITY** - Document is uploaded appropriately

~~*Repeat this test by dragging additional documents after the single file~~
~~*Repeat this test by uploading a single file, then uploading multiple~~
~~*Repeat this test by attempting to upload more than 5 documents - component should limit it to 5~~



#### 5) Petitioner can remove a file that was added in the previous step

- Continuing from the previous test
- Remove the file ~~one, some and all files~~ that ~~were~~  was added in the IRS Notice section

Expected Results
- [x] **NEW FUNCTIONALITY** - The file can be removed, and the user can ~~drag/~~ upload another document if desired.



#### 6) Petitioner continues the Petition flow, reviews their case before submitting, IRS notice~~s added are~~ is displayed individually for preview on the Review page.

- Continuing from the previous test
- Be sure that you have at least one IRS Notice attached
- Click Continue to Step 3 of 5 button
- Fill in the required information about the Petitioner
- Click Continue to Step 4 of 5 button
- Select a Case Procedure and Trial location
- Click Continue to Step 5 of 5 button
- Review the "Review your Case" screen

Expected Results
- [x] **NEW FUNCTIONALITY** - New IRS Notice label is under the Petition label on the About Your Case card.
- [x] **NEW FUNCTIONALITY** - The IRS Notice~~s~~ that you uploaded ~~are~~ is displayed as a link~~s~~ under the IRS Notice Label
- [x] **NEW FUNCTIONALITY** - Clicking on the link displays the Notice in a Preview modal (the same as how we currently display the Petition for review).
- [x] **NEW FUNCTIONALITY** - The Service email is now underneath the Petitioner Contact information; There is no longer an entire section devoted to this data.
- [x] **NEW FUNCTIONALITY** - The information in the "A few reminders before you submit" section has been updated and matches the mockup.



#### 7) Petitioner Submits the Case to the US Tax Court; IRS notice~~s~~ that ~~were~~ was added ~~are~~ is reflected as "Attachments to Petition" on the docket record and ~~are~~ is not yet served.

- Continuing from the previous test
- Click on the Submit to U.S. Tax Court button
- Click on the Case Docket number after it is submitted
- Review the Docket Record

Expected Results:
- [x] **NEW FUNCTIONALITY** - The IRS Notice ~~(s)~~ that you submitted ~~are~~ is displayed on the docket record
- [x] **NEW FUNCTIONALITY** - None of the documents are served yet
- [x] **NEW FUNCTIONALITY** - Event code for the IRS notice is ATP



#### 8) Petitioner starts a case; Indicates that they did not receive a notice; no section displays to drag/upload any PDF's.

- Log in as a Petitioner
- Click on Create Case
- Fill in the required fields
- On the IRS Notice section, select No, that you did not receive a notice from the IRS

Expected Results:
- [x] There is no section to drag or upload files.
- [x] Completing the petition flow without an upload works as per usual, and Petition submits successfully.



#### 9) Petitioner starts a case; Indicates that they DID receive an IRS notice; Petitioner does NOT upload or drag any documents; Petition submits successfully with no validation errors.

- Log in as a Petitioner
- Click on Create Case
- Fill out the required fields
- On the IRS notice section, select yes, that you did receive a notice from the IRS
- Do not ~~drag or~~ upload any PDF
- Continue with the workflow and fill in the required fields

Expected Results:
- [x] **NEW FUNCTIONALITY** - There are no validation error messages if the user does not upload any IRS notice, and the Petition submits as per normal



#### 10) Petitioner uploads a PDF that is non-compliant (PW protected/encrypted/non-PDF file, over 250 MB etc.) for the IRS notice; User receives Validation Error message upon submitting to USTC.

- Log in as a Petitioner
- Click on Create a Case
- Fill out the required fields
- On the IRS notice section, upload ~~1-5 documents~~ a document, making sure that ~~at least one of them~~ it is non-compliant.
- Continue with the workflow by filling out all of the required fields
- Click on the Submit to US Tax Court

Expected Results:
- [x] **NEW FUNCTIONALITY** - User receives the standard "There was a problem uploading your documents" modal
- [x] **NEW FUNCTIONALITY** - User can go back through the workflow to remove and re-add the appropriate document and then submit successfully.



#### 11) Petitions Clerk QC's a Petition with an IRS notice that was uploaded; New tab in the Document viewer that displays ATP.

- Log in as a Petitions Clerk
- Click on Document QC
- Click on Section Document QC
- Find the Petition that was submitted with one IRS Notice from the tests above
- Click on the Petition Link

Expected Results:
- [x] **NEW FUNCTIONALITY** - There is now a new tab on the Add Document(s) section that displays ATP



#### 12) Petitions Clerk views the ATP tab, The IRS Notice that was uploaded in the previous tests is displayed in the document viewer.

- Continuing from the previous test
- Click on the ATP Tab in the Add Document(s) Section

Expected Results:
- [x] **NEW FUNCTIONALITY** - The IRS Notice that was uploaded in the previous tests displays in the document viewer.



#### 13) Petitions Clerk completes the QC; IRS Notice is available to preview on the Review and Serve Petition Page

- Continuing from the previous test
- Be sure that the required fields are all filled in on the Petition
- Click on the Review Petition button

Expected Results
- [x] **NEW FUNCTIONALITY** - The IRS Notice that was uploaded in the previous tests is displayed as a link on the Attachments Card
- [x] **NEW FUNCTIONALITY** - After clicking on the link, the IRS Notice is displayed in a document preview pop-up modal
- [x] **NEW FUNCTIONALITY** - The IRS Notice in the pop-up modal has a coversheet appended to the upload with the usual info (case caption, received date, filed date, "Electronically filed", Docket number, document number, title is Attachment to Petition), but no served stamp at the bottom of the coversheet yet.
- [x] **NEW FUNCTIONALITY** - Clicking closed from the modal closes the preview



#### 14) Petitions Clerk serves the petition to the IRS, ATP document is reflected as served on the docket record, Service email is sent appropriately.

- Continuing from the previous test
- Click on the Serve to IRS button
- Click Yes, Serve button
- Click on the Served tab
- Click on the docket number of the Petition you just served

Expected Results:
- [x] **NEW FUNCTIONALITY** - The IRS Notice is on the docket record and has a service date of today's date
- [x] **NEW FUNCTIONALITY** - The IRS Notice Service parties code is R
- [x] **NEW FUNCTIONALITY** - Kibana logs indicate successful service of the ATP to the IRS Service Agent



~~#### 15) Repeat steps 11-14 with a Petition that was filed that had more than 1 IRS Notice attached.~~

~~Expected Results:~~
~~- [ ] **NEW FUNCTIONALITY** - All tests from above pass, but instead of one tab in the Add Document(s) section for the ATP, there are multiple tabs displaying ATP for each document uploaded.~~
~~- [ ] **NEW FUNCTIONALITY** - On service of Petition the IRS, All ATP documents are served individually, service party code for each document is R, Kibana logs display successful service to IRS service agent.~~
 


#### 16) Petitions clerk QC's a petition that has an ATP by clicking on the Docket number link from QC, then on the Review and Serve Petition Link; 

**Part 1**
- Log in as a Petitions Clerk
- Click on Document QC
- Click on Section Document QC
- Click on the Docket number link (not the Petition Link) from the QC list
- Click on one of the ATP document links

Expected Results:
- [x] **NEW FUNCTIONALITY** - There is no link in the document viewer to Review and Serve Petition

**Part 2**
- Click on the Petition document link or the Petition from the document view tab 
- Click on the Review and Serve Petition link in the document viewer

Expected Results:
- [x] **NEW FUNCTIONALITY** - There is a new tab in the "Add Document(s)" document viewer for the ATP. ~~If multiple were uploaded in this case, there are individual ATP tabs displayed.~~
- [x] **NEW FUNCTIONALITY** - ~~Each~~ The ATP that was uploaded is displayed and has a coversheet with the appropriate info (case caption, received date, filed date, docket number, title, etc.)

**Part 3**
- Click Review Petition button

Expected Results:
- [x] **NEW FUNCTIONALITY** - The IRS Notice ~~(s)~~ that ~~were~~ was attached by the petitioner ~~are~~ is displayed as a hyperlink in the Attachments Card
- [x] **NEW FUNCTIONALITY** - Petitions clerk can click on the link and a pop-up modal document preview displays the correct IRS Notice.

**Part 4** 
- Click close on the document viewer
- Click Serve to IRS
- Click Yes, Serve
- User is routed to the document view on the case
- Click on the docket record tab

Expected Results:
- [x] **NEW FUNCTIONALITY** - The IRS Notice ~~s are~~ is reflected on the docket record and ~~are~~ is titled "Attachment to Petition" with a Service date of today's date, served parties code is R.
- [x] **NEW FUNCTIONALITY** - Kibana logs indicate successful service of ~~each~~ the ATP to the IRS Service Agent.



#### 17) Petitions Clerk begins QC of an electronically submitted Petition that has an ATP; Saves the Petition for later QC; Petition and ATP are served appropriately after going back to the QC item in Progress.

- Log in as a Petitions Clerk
- Click on Document QC
- Click on Section Document QC
- Click on the Petition link to begin QC
- Ensure that there is an ATP tab(s).
- Click Review Petition button
- Click on the Save for later button (do not serve to IRS)
- Click on the In Progress tab from the Document QC tab
- Click on the Petition link from the In Progress tab
- Verify that you can click on the edit links from the review screen and the ATP tab displays in the Add Document(s) viewer.
- Click on Review petition link
- Click on Serve to IRS
- Click Yes, Serve
- Click on the Served tab
- Click on the Docket number link of the Petition that you just served

Expected Results:
- [x] **NEW FUNCTIONALITY** - The IRS Notice ~~s are~~ is reflected on the docket record and ~~are~~ is titled "Attachment to Petition" with a Service date of today's date, served parties code is R.
- [x] **NEW FUNCTIONALITY** - Kibana logs indicate successful service of ~~each~~ the ATP to the IRS Service Agent.



#### 18) Repeat the steps from Test 11 - 17 with the Case Services Supervisor Role and the Cotc role.

- [x] All tests pass



#### 19) Regression test ATP document visibility

- [x] ATP document is not available for the public non-logged in user to view on the docket record
- [x] ATP document is viewable by parties on the case
- [x] ATP document is viewable by the Terminal user 
- [x] ATP document can be sealed by a docket clerk - when sealed, document is not visible by the terminal user, and is not viewable by parties if it sealed to the public and parties of the case.

**New test case added 2/29/24**

#### 20) Test the external workflow on mobile devices

- [x] Design looks agreeable on mobile screens
- [ ] Functionality works the same as it does on desktop

**Comment by cholly75** [comment]
@jasonfu9966 @nechama-krigsman @Absolutestunna @ttlenard @katiecissell - please leave the petition upload download component as-is for this story. We will look at rationalizing the upload components in another story specifically or perhaps incorporate into ustaxcourt/ef-cms#7175.

**Comment by jasonfu9966** [comment]
Two additional pieces of functionality that should be covered in testing:
- a petitioner should be able to remove ATPs from the multi-upload component by clicking "change files" and then "cancel"
- if a petitioner uploads ATP files and proceeds to the next step (or back) and returns to the step with the ATP upload component, that component should show the files that were previously uploaded

**Comment by jasonfu9966** [comment]
The Figma file for this story includes a modal that tells the user that their file size is too big, with an email link at the bottom of it. 

After discussing with UX, we decided to table the inclusion of that modal to the new petitions e-filing flow and leave this component's alert as a javascript alert. This is so the three file upload components in this particular flow have the same "file is too large" alert.

**Comment by swongCO** [comment]
**UX Review**
- new upload component: Remove File link persists after file is removed by clicking Cancel in the file selector
<img width="604" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBemhJQXc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--f78adc0b82adf4596705e69b90ae63e53e2ede3f/Screenshot%202024-02-09%20at%209.50.25%E2%80%AFAM.png" alt="Screenshot 2024-02-09 at 9.50.25 AM.png" />
- as a case services supervisor QCing a petition, I get a 404 when clicking on the STIN tab; this behavior is different on staging and IRS envs; change the behavior on test so it mimics the other envs (no 404, instead it allows a STIN to be uploaded or scanned)
- as a petitioner, when I upload a password protected IRS notice, when I submit, it shows a modal indefinitely (should it be showing the second error modal?)
<img width="643" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBenBJQXc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--9275c34e1cfd8aa861d345c3433c97d85ad06855/Screenshot%202024-02-09%20at%2011.15.55%E2%80%AFAM.png" alt="Screenshot 2024-02-09 at 11.15.55 AM.png" />
<img width="525" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBejVJQXc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--b2b2e7220029e5450b2b8ff993acb850819688c8/Screenshot%202024-01-25%20at%209.43.47%E2%80%AFAM.png" alt="Screenshot 2024-01-25 at 9.43.47 AM.png" />

**Comment by ttlenard** [comment]
@Absolutestunna @nechama-krigsman @jasonfu9966 Some preliminary Testing Feedback. I'll continue testing but wanted to get you this info.

Caseless Petitioner page has some differences from the Mock:
<img width="676" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBMHhoQXc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--750a58d9b62c76ab9cd0be309de1dfa0a381fd57/image.png" alt="image.png" />


Accordion menu has some differences from mock:
<img width="1118" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBMUZoQXc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--73209cb92f5064bee44e4417833915e287e30950/image.png" alt="image.png" />


The upload component has some funky behavior.  I can demo this for you if the explanation below isn't clear.
1. If you drag files over, and grab 3, and then go back to your folder and drag some more, the previous documents that you dragged are replaced. You should be able to drag additional without replacing, correct?
2. Similar behavior if you choose from folder. If you select one document, and then click within the document area, you get the folder again, but once you select another document, the first document disappears.
3. Not sure if this is an issue, but if you include a file that is over 250 MB, the entire cluster of documents that you attempt to bring in will clear out.

**Comment by swongCO** [comment]
Discussion from today: remove the pdf icon from the petition in the Review page. There should be no pdf icons in front of any links.
Upload component TBD until we talk to Chris.

**Comment by ttlenard** [comment]
Additional feedback: I made a video of some behavior with the upload progress bar that I am seeing when a petitioner submits a case to the Court with the ATP story.  This happens if you have largish documents (I've tested with docs that are 2-24MB, and 24-216MB) that you have uploaded. I don't see this on IRS, or Staging.  Thoughts on user experience? 
[Private Zenhub Video](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBNjltQXc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--f4fa3ec9e9ef9c2c76004c7f67ed711ef133f45d/Submit%20to%20USTC%20progress%20bar.mp4)

**Comment by jasonfu9966** [comment]
After much discussion, this ensemble has made the difficult decision to revert the work done on the new file input component and use the old one instead. 

This is due to the requirement changing from accepting multiple ATP files to accepting just one (which the old component handles with confidence), and because of the need for a better multi-file upload component than USDWS currently offers. We would like to create a custom one of our own.

We aren't sure when the new input will be developed, but the tentative plan for it is to be introduced some time between now and the 'revamped' electronic petition flow.

**Comment by ttlenard** [comment]
@jasonfu9966  @Absolutestunna @nechama-krigsman  I've created another video of some behavior I am seeing with the progress bar. I noticed that it is going beyond 100%.  I think I've seen this before in other areas of the application? Maybe it was @codyseibert that looked into it?

[Private Zenhub Video](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBL3FlQXc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--940b103067f46402414a6cb00029fa9b7c362950/progress%20bar.mp4)
